src/botPage/view/blockly/index.js

```diff
@@ -317,8 +317,40 @@ export default class _Blockly {
     load(blockStr = '', dropEvent = {}) {
         let xml;
 
+        try {
+            const xmlDoc = new DOMParser().parseFromString(blockStr, 'application/xml');
+
+            if (xmlDoc.getElementsByTagName('parsererror').length > 0) {
+                globalObserver.emit('ui.log.warn', `${translate('Unrecognized file format')}`);
+                return;
+            }
+        } catch (err) {
+            globalObserver.emit('ui.log.warn', `${translate('Unrecognized file format')}`);
+            return;
+        }
+
         try {
             xml = Blockly.Xml.textToDom(blockStr);
+            const blocklyXml = xml.querySelectorAll('block');
+
+            if (blocklyXml.length <= 0) {
+                globalObserver.emit(
+                    'ui.log.warn',
+                    `${translate(
+                        'Unsupported strategy. Binary Bot only accepts XML scripts from Binary.com sources. Please load a supported file.'
+                    )}`
+                );
+                return;
+            }
+
+            blocklyXml.forEach(block => {
+                const blockType = block.attributes.type.nodeValue;
```

Comment:
Probably better to use `getAttribute` here as it's clearer, and because the `type` attribute on the `block` element is a custom attribute, in some browsers this custom attribute may not be synced with the DOM object. 

```suggestion
                const blockType = block.getAttribute('type');
```